playwright-gh-pages/README.md

```diff
@@ -0,0 +1,223 @@
+
+      - name: Wait for grafana server
+        uses: grafana/plugin-actions/wait-for-grafana@main
+
+      - name: Install Playwright Browsers
+        run: npm exec playwright install chromium --with-deps
+
+      - name: Run Playwright tests
+        id: run-tests
+        run: npm run e2e
+
+      # use upload-report-artifacts Action to upload the report and the test summary to GH Artifacts
+      - name: Upload e2e test summary
+        uses: grafana/plugin-actions/playwright-gh-pages/upload-report-artifacts@main
+        if: ${{ (always() && !cancelled()) }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          test-outcome: ${{ steps.run-tests.outcome }}
+
+  deploy-pages:
+    if: ${{ (always() && !cancelled()) }}
+    needs: [playwright-tests]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # use deploy-report-pages Action to deploy the artifacts to GitHub Pages
+      - name: Publish report
+        uses: grafana/plugin-actions/playwright-gh-pages/deploy-report-pages@main
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+```
+
+### Example using a per-plugin matrix
+
+The following simplified example demonstrates how Playwright report publishing can be integrated in a mono repo where the matrix is derived for each plugin in the repo.
+
+```yaml
+name: e2e tests
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+
+permissions:
+  contents: write
+  id-token: write
+  pull-requests: write
+
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        plugin-id: ['grafana-panel-sample1', 'grafana-panel-sample2', 'grafana-panel-sample3']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # necessary steps to install dependencies and build the plugin
+
+      - name: Start Grafana latest
+        run: |
+          docker compose pull
+          DEVELOPMENT=false GRAFANA_VERSION=latest GRAFANA_IMAGE=grafana-enterprise docker compose up -d
+
+      - name: Wait for grafana server
+        uses: grafana/plugin-actions/wait-for-grafana@main
+
+      - name: Install Playwright Browsers
+        run: npm exec playwright install --with-deps
+
+      - name: Run Playwright tests
+        id: run-tests-latest
+        run: npm run e2e
+
+      # use upload-report-artifacts Action to upload the report and the test summary to GH Artifacts
+      - name: Upload e2e test summary
+        uses: grafana/plugin-actions/playwright-gh-pages/upload-report-artifacts@main
+        if: ${{ (always() && !cancelled()) }}
+        with:
+          report-dir: playwright-report
+          grafana-version: latest
+          grafana-image: grafana-enterprise
+          plugin-name: ${{ matrix.plugin-id }}
+          test-outcome: ${{ steps.run-tests-latest.outcome }}
+      # repeat steps but for another Grafana version if necessary
```

Comment:
not quite clear why it would be necessary, isn't it an example for matrix test with multiple grafana versions? (from mobile so maybe did not look it up properly)

Reply:
No for this example, the matrix resolves multiple plugins. Could for example be in a monorepo where you repeat per plugin. Not sure how common it would be, but basically just want to highlight that it can be used in many different ways.